src/main/java/uk/ac/ebi/eva/commons/core/models/genotype/Genotype.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.commons.core.models.genotype;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+
+
+public class Genotype {
+
+    private String reference;
+    private String alternate;
+    private int[] allelesIdx;
+    private boolean phased;
+
+    private AllelesCode code;
+
+    private int count;
+    protected static final Pattern genotypePattern = Pattern.compile("/|\\|");
+
+
+    Genotype() {
+    }
+
+    public Genotype(String genotype) {
+        this(genotype, null, null);
+    }
+
+    public Genotype(String genotype, String ref, String alt) {
+        this.reference = ref;
+        this.alternate = alt;
+        this.phased = genotype.contains("|");
+        this.count = 0;
+        parseGenotype(genotype);
+    }
+
+
+    private void parseGenotype(String genotype) {
+        String[] alleles = genotypePattern.split(genotype, -1);
+
+        this.code = alleles.length > 1 ? AllelesCode.ALLELES_OK : AllelesCode.HAPLOID;
+        this.allelesIdx = new int[alleles.length];
+
+        for (int i = 0; i < alleles.length; i++) {
+            String allele = alleles[i];
+
+            if (allele.equals(".") || allele.equals("-1")) {
+                this.code = AllelesCode.ALLELES_MISSING;
+                this.allelesIdx[i] = -1;
+            } else {
+                if (StringUtils.isNumeric(allele)) { // Accepts genotypes with form 0/0, 0/1, and so on
```

Comment:
Can be replaced with Guava's method [Ints#tryParse(String)](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/primitives/Ints.html#tryParse-java.lang.String-) by doing:

```java
if (Ints.tryParse(allele) != null) {
    ...
}
```